shared/proc/CellClassification/CategorizeStriatumWave.m

```diff
@@ -0,0 +1,135 @@
+function out = CategorizeStriatumWave(cfg_in,S)
+% function out = CategorizeStriatumWave(cfg,S)
+%
+% striatal data cell type classification based on waveform and firing rate
+% following Berke et al. Neuron 2004
+%
+% NOTE: *wv and *ClusterQual files MUST be present (use Create_CQ_File.m)
+%
+% INPUTS:
+%
+% S: ts struct with spike data
+%
+% CONFIGS:
+%
+% cfg.MinID = 20; % minimum isolation distance for cell to be considered
+% cfg.MaxLr = 0.1; % maximum L-ratio for cell to be considered
+% cfg.MinAmpl = 200; % minimum peak (max) amplitude for cell to be considered
+% cfg.cMethod = 'Berke'; % selects set of classification criteria to use
+%   (alternate: 'MvdM', i.e. as in van der Meer & Redish, Front Int Nsci 2009)
+%   (altnerate: 'Sirota', i.e. as in Karalis & Sirota, 2018) note:
+%   currently only uses valley-to-peak time and not asymmetry index
+%
+% OUTPUTS:
+%
+% out.msn: idx into S.t for cells classified as putative medium spiny
+%   neuron
+% out.fsi: idx into S.t for cells classified as putative fast-spiking
+%   interneuron
+% out.other: idx into S.t for cells classified as other (neither MSN not
+%   FSI)
+% out.wv: waveforms for each neuron (taken directly from *wv files)
+% out.cq: cluster quality metrics for each neuron (taken directly from *cq
+%   files)
+% out.fr: firing rate (mean(1/ISI)) for each neuron
+% out.pw: waveform peak width (ms) for each neuron
+% out.vw: waveform valley width (ms) for each neuro
+%
+% MvdM 2015-02-05 
+
+cfg_def = [];
+cfg_def.MinID = 20;
+cfg_def.MaxLr = 0.1;
+cfg_def.MinAmpl = 600;
+cfg_def.cMethod = 'Berke'; % alt: 'MvdM'
+
+cfg = ProcessConfig(cfg_def,cfg_in);
+
+
+nCells = length(S.t);
+for iC = nCells:-1:1
+
+    cellfname = S.label{iC};
+
+    clufname = regexprep(cellfname,'_','-');
```

Comment:
This is OK for now, but this block of code accomplishes a specific function -- loading the WV and CQ files -- that could be neatly outsourced into a local function. Then, reading this function will be easier.